x-pack/plugins/apm/public/context/url_params_context/url_params_context.tsx

```diff
@@ -60,7 +58,8 @@ const UrlParamsProvider: React.ComponentClass<{}> = withRouter(
 
     const { start, end, rangeFrom, rangeTo } = refUrlParams.current;
 
-    const [, forceUpdate] = useState('');
+    // Counter to force an update in useFetcher when the refresh button is clicked.
+    const [rangeId, setRangeId] = useState(0);
```

Comment:
We already have a `setCounter` method in useFetcher that's invoked when calling `refetch()`. Do we need both?

Reply:
One is a mechanism to identify when the url changes, the other is a mechanism to refetch upon an action. We might need both but perhaps we can them improve the names so they indicate what they each do?

Reply:
I think we can use `counter` for both cases. I'll take a look.

Reply:
`refetch` and `refreshTimeRange` are used differently like you said, and we're not using `useFetcher` in the datepicker so don't have access to a counter there. I think this is sufficient as-is but if we want to further refactor and clarify later that would be good.